src/Closure.cpp

```diff
@@ -106,5 +109,79 @@ void Closure::visit(const Atomic *op) {
     op->body.accept(this);
 }
 
+Expr Closure::pack_into_struct() const {
+    std::vector<Expr> elements;
+    std::vector<Type> types;
+
+    std::ostringstream name;
+
+    for (const auto &b : buffers) {
+        Expr ptr_var = Variable::make(type_of<void *>(), b.first);
+        elements.emplace_back(ptr_var);
+    }
+    for (const auto &v : vars) {
+        Expr var = Variable::make(v.second, v.first);
+        elements.emplace_back(var);
+    }
+
+    // Sort by decreasing size, to guarantee the struct is densely packed in
+    // memory. We don't actually rely on this, it's just nice to have.
+    std::stable_sort(elements.begin(), elements.end(),
+                     [&](const Expr &a, const Expr &b) {
+                         return a.type().bytes() > b.type().bytes();
+                     });
+
+    for (Expr e : elements) {
```

Comment:
style nit: declaring `types` and `name` here (instead of at top) is perhaps better.

Reply:
This was dead code. I deleted it.